pkg/controller/network_policy.go

```diff
@@ -558,13 +620,46 @@ func (c *Controller) fetchPolicySelectedAddresses(namespace, protocol string, np
 			for _, podIP := range pod.Status.PodIPs {
 				if podIP.IP != "" && util.CheckProtocol(podIP.IP) == protocol {
 					selectedAddresses = append(selectedAddresses, podIP.IP)
+
+					svcs, err := c.servicesLister.Services(pod.Namespace).List(labels.Everything())
+					if err != nil {
+						klog.Errorf("failed to list svc, %v", err)
+						return nil, nil, fmt.Errorf("failed to list svc, %v", err)
+					}
+					klog.Infof("svc is %v", svcs)
+					svcIps := svcMatchPods(svcs, pod)
+					klog.Infof("svcIps is %v", svcIps)
+					selectedAddresses = append(selectedAddresses, svcIps...)
 				}
 			}
 		}
 	}
 	return selectedAddresses, nil, nil
 }
 
+func svcMatchPods(svcs []*corev1.Service, pod *corev1.Pod) []string {
+	match := []string{}
+       // find  svc ip by pod's info
+	for _, svc := range svcs {
+		if isSvcMatchPod(svc, pod) && svc.Spec.ClusterIP != "None" &&
+			svc.Spec.ClusterIP != "" {
+			match = append(match, svc.Spec.ClusterIP)
+		}
+	}
+	return match
+}
+func isSvcMatchPod(svc *corev1.Service, pod *corev1.Pod) bool {
+	ss := metav1.SetAsLabelSelector(svc.Spec.Selector)
+	sel, _ := metav1.LabelSelectorAsSelector(ss)
+	if pod.Labels == nil {
+		pod.Labels = map[string]string{}
+	}
```

Comment:
`pod` should be read only, so we should return false directly here.

Reply:
ok